### PR TITLE
FEDX-13: Dart 2.18 lints & formatting

### PR DESCRIPTION
**Pull request by evanweible-wf**
# [FEDX-13](https://jira.atl.workiva.net/browse/FEDX-13)
Issue Status

- Address new lints on 2.18
- Run formatter on 2.18
- Replace deprecated pedantic package with lints package
- Tweaks to CI

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by matthewnitschke-wk — APPROVED** [review]

File: pubspec.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies:
 
 dev_dependencies:
   dependency_validator: ^3.0.0
+  lints: ^1.0.1
```
[the remark this answers is not in the document]
Comment (evanweible-wf) [reply pubspec.yaml:26]: Unfortunately not quite :( We use `unawaited` and that wasn't added to `dart:async` until a version of Dart after 2.11, so we'd have to raise our minimum. I plan to drop it after the NNBD migration.

[the remark this answers is not in the document]
Comment (robbecker-wf) [reply pubspec.yaml:26]: ok, sounds great.

**Review by robbecker-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
QA +1 CI passes
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**